packages/ai-ide/src/common/summarize-session-commands.ts

```diff
@@ -0,0 +1,22 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { Command } from '@theia/core';
+
+export const AI_SUMMARIZE_SESSION_AS_TASK_FOR_CODER: Command = {
```

Comment:
Going with a command even though I set up the callback system!? ;-)

Reply:
I did this, because I wanted to have the nice progress bar, so it felt natural. I would use call back for actions that stay within the session for example pre-defined user message. Does this make sense to you?

Reply:
The command approach was my preference, so I'm happy either way. Re: the progress bar, its display is part of the behavior of `TaskContextService.summarize`, so as long as you call that, whether in a command or a callback, you should get the progress bar.

Reply:
I think it is good to support both!

Do you have any other comments on this PR?

Reply:
No - not sure if it makes sense to approve it while #15427 is still pending? Or would you like this to go in instead?